src/main.go

```diff
@@ -79,53 +73,91 @@ type DfoApiUrlObj struct {
 	Url         string
 }
 
+// Input variables in order they are requested
+type InputDataObj struct {
+	BusNo        string
+	ClientId     string
+	ClientSecret string
+	Cluster      string
+	DateFrom     string
+	DateTo       string
+	Env          string
+	Region       string
+	SortType     string
+	TenantId     string
+}
+
+type TenantIdsObj struct {
+	TenantIDs [1]string `json:"tenantIds"`
+}
+
 func main() {
-	var dfoData []models.DataView
-	var dfoDataErr models.DataView
-	var dmwKnownContacts models.DmwKnownContacts
-	// Input variables in order they are requested
-	region, env, tenantId, busNo, clientId, clientSecret, dateFrom, dateTo, sortType := "", "", "", "", "", "", "", "", ""
+	var dfoDataList []models.DataView
+	var dfoData models.DataView
+	var dmwKnownContact models.DmwKnownContacts
+	var inputData InputDataObj
+	var log []byte
+	var logMsg string
 	st := time.Now()
 	var tenants [1]string
 	var wg sync.WaitGroup
 
 	//Prompt for needed input data
-	region = promptForInputData("region", regionRequest)
-	env = promptForInputData("env", envRequest)
-	tenantId = promptForInputData("tenantId", tenantGuidRequest)
-	busNo = promptForInputData("busNo", busNoRequest)
-	clientId = promptForInputData("clientCreds", clientIdRequest)
-	clientSecret = promptForInputData("clientCreds", clientSecretRequest)
-	dateFrom = promptForInputData("dateFrom", dateFromRequest)
-	if dateFrom != "" {
-		dateTo = promptForInputData("dateTo", dateToRequest)
-		sortType = promptForInputData("sortType", sortTypeRequest)
+	inputData.Region = promptForInputData("region", regionRequest)
```

Comment:
if I do NA1 instead of na1, will that be safe?  I'm assuming this will be for building the full URL to communicate with DFO's API, right?  If so, saul good, man and no normalization needed.

Reply:
yes - I take the input string and do strings.ToLower(inputValue).  Yes, this was for building the full URL.  If we are running this inside dmw, then the way we build the URL can get the values from the terraform variables rather than asking for input data.